src/charm.py

```diff
@@ -301,24 +289,22 @@ def _deploy_k8s_resources(self) -> None:
         self.model.unit.status = MaintenanceStatus("K8S resources created")
 
     def _on_install(self, _):
-        """Perform installation only actions."""
-        self._check_container_connection()
-
-        # proceed with the same actions as for Pebble Ready event
-        self._on_pebble_ready(_)
+        """Installation only tasks."""
```

Comment:
I believe you need the cert files in the container to perform the installation correctly, which requires the container to be up and ready to push the files, which requires pebble to be ready also.

Did you try running inference or some small workload to confirm the above is not required?

Reply:
Install event happens before Pebble Ready event. At the time of Install event container is most likely is not ready, so we just deploy K8S resources in `_on_install` handler to speed up deployment and do nothing else. If we try to upload certificates to container in `_on_install`, it will fails if container is not up (that's why there is container connection check in upload function). Container might be up, but it is not guaranteed, that's what Pebble Ready event is for - indicate that container has been seen.
In `_on_pebble_ready` handler we check for connection to container and upload certificates (yes, there is chance that container can go down between the check and upload, but that's the framework we are working with).

Reply:
I understand the process and why you removed some methods from the install event handler, I am wondering if by doing this we are ditching the cert files that are required to correctly configure the k8s resources you are applying on_install.

Reply:
Closing it as per our discussion, our integration tests should be robust enough to catch any irregular behaviour. If this is an issue in production, let's raise it again.